scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/NameDetectAnalyzer.kt

```diff
@@ -0,0 +1,268 @@
+private const val NMS_THRESHOLD = 0.75F
+private const val CHAR_CONFIDENCE_THRESHOLD = 0.7
+
+
+class NameDetectAnalyzer private constructor(
+    private val ssdObjectDetect: SSDObjectDetect?,
+    private val alphabetDetect: AlphabetDetect?
+) : Analyzer<SSDOcr.SSDOcrInput, PaymentCardOcrState, NameDetectAnalyzer.Output> {
+
+    data class Output(
+        val name: String?,
+        val boxes: List<DetectionBox>?
+    )
+
+    override val name: String = "card_side_analyzer"
+
+    override suspend fun analyze(data: SSDOcr.SSDOcrInput, state: PaymentCardOcrState): Output {
+        return if (state.runNameExtraction && ssdObjectDetect != null && alphabetDetect != null) {
+            val objDetectInput = SSDObjectDetect.Input(
+                fullImage = data.fullImage,
+                previewSize = data.previewSize,
+                cardFinder = data.cardFinder,
+                iin = null
+            )
+
+            val output = ssdObjectDetect.analyze(objDetectInput, Unit)
+            var nameRect: RectF? = null
+
+            // check to see if we detected a legal name. If so, run our name extraction model, serially for now
+            for (box in output.detectionBoxes) {
+                if (box.label == SSDObjectDetect.Labels.NAME.ordinal) {
+                    nameRect = box.rect
+                    break
+                }
+            }
+
+            val name = if (nameRect != null) {
+                processPredictions(nameRect, objDetectInput.toObjectDetectionCroppedImage())
+            } else {
+                null
+            }
+
+            Output(name, output.detectionBoxes)
+        } else {
+            Output("", null)
+        }
+    }
+
+    internal class CharPredictionWithBox(
+        val characterPrediction: AlphabetDetect.Prediction,
+        val box: RectF
+    ) {
+        private val boxArray: FloatArray = FloatArray(4)
+
+        fun getPercentBoxArray(width: Int, height: Int): FloatArray {
+            val pctBoxArray = FloatArray(4)
+            pctBoxArray[0] = boxArray[0] / width
+            pctBoxArray[1] = boxArray[1] / height
+            pctBoxArray[2] = boxArray[2] / width
+            pctBoxArray[3] = boxArray[3] / height
+            return pctBoxArray
+        }
+
+        init {
+            boxArray[0] = box.left
+            boxArray[1] = box.top
+            boxArray[2] = box.right
+            boxArray[3] = box.bottom
+        }
+    }
+
+    private suspend fun processPredictions(
+        nameRect: RectF,
+        bitmapForObjectDetection: Bitmap
+    ): String? {
+        if (alphabetDetect == null) {
+            return null;
+        }
+        var x = (nameRect.left * bitmapForObjectDetection.width).toInt()
+        val y = (nameRect.top * bitmapForObjectDetection.height).toInt()
+        var width = (nameRect.width() * bitmapForObjectDetection.width).toInt()
+        val height = (nameRect.height() * bitmapForObjectDetection.height).toInt()
```

Comment:
since we're using this logic in a few places, maybe we should extract it to a helper function?